gammapy/maps/wcsnd.py

```diff
@@ -589,6 +604,42 @@ def get_spectrum(self, region=None, func=np.nansum, weights=None):
 
         return self.to_region_nd_map(region=region, func=func, weights=weights)
 
+    def binary_erode(self, width, from_edges=False):
+        """Binary erosion of boolean mask removing a given margin
+        Parameters
+        ----------
+        width : tuple of `~astropy.units.Quantity`
+            Angular sizes of the margin in (lon, lat) in that specific order.
+            If only one value is passed, the same margin is applied in (lon, lat).
+        """
+        mask_data = _apply_binary_operations(self, width, ndi.binary_erosion)
+        return self._init_copy(data=mask_data)
+
+    def boundary_mask(self, width):
```

Comment:
For consistency please move this to `WcsGeom.boundary_mask(width, inside=True)`. We already have `WcsGeom.region_mask()` which does a very similar thing. Internally you can still rely on the `.binary_erode()`, however there might be more efficient implementations...

Reply:
-moved .boundary_mask() from WcsNDMap to WcsGeom
-modified WcsGeom.region_mask() such as output is a WcsNDMap

Reply:
Perfect thanks! In general it's maybe not a bad idea to have the mask creation on the `Map` level, because we have the pattern of "factory methods", such as `.create()`, `.from_geom()`, etc., but for now the consistency is more important...